### PR TITLE
Refactor computepoipoints

### PR DESCRIPTION
**Pull request by magangloff**
- Refactored ComputePOIPoints. There is now a POIPointsUtilities class.
- Added ZoomOnUserLocation and changeMapTile button to the download map.
- Fixed bug #236

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/sdp/peakar/general/SettingsMapActivityTest.java [hunk 1]
```diff
@@ -181,4 +193,39 @@ private void readFromFile() {
...
+
+    /* Check that a press on "zoom on user location button effectively zooms on user loc"*/
+    @Test
+    public void pressZoomOnUserLocationButton() throws InterruptedException {
+        OSMMap osmMap = SettingsMapActivity.osmMap;
+        MapView mapView = osmMap.getMapView();
+        //Get map center at start
+        GeoPoint geoPointStart = (GeoPoint) mapView.getMapCenter();
+        //Wait 5 Sec for the provider to get the location
+        Thread.sleep(5000);
```
Comment (bgabriel998) [inline app/src/androidTest/java/ch/epfl/sdp/peakar/general/SettingsMapActivityTest.java:222]: Use thread sleep constant instead. (the suggested constant needs to be added + a reference to TestingConstant file in the imports)
```suggestion
        Thread.sleep(THREAD_SLEEP_5S);
```

**Review by bgabriel998 — COMMENTED** [review]
**Remarks:**
In **SettingsMapActivityTest.java**  the variable ret is assigned but _never used_. Can it be removed?

File: app/src/androidTest/java/ch/epfl/sdp/peakar/general/SettingsMapActivityTest.java [hunk 2]
```diff
@@ -19,21 +19,33 @@
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
+import org.osmdroid.tileprovider.MapTileProviderBase;
+import org.osmdroid.util.GeoPoint;
 import org.osmdroid.views.MapView;
+import org.osmdroid.views.overlay.Overlay;
+import org.osmdroid.views.overlay.mylocation.MyLocationNewOverlay;
```
Comment (ravifrancesco) [inline app/src/androidTest/java/ch/epfl/sdp/peakar/general/SettingsMapActivityTest.java:26]:
```suggestion
```

File: app/src/androidTest/java/ch/epfl/sdp/peakar/general/SettingsMapActivityTest.java [hunk 3]
```diff
@@ -19,21 +19,33 @@
...
 
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.InputStreamReader;
+import java.util.List;
```
Comment (ravifrancesco) [inline app/src/androidTest/java/ch/epfl/sdp/peakar/general/SettingsMapActivityTest.java:32]:
```suggestion
```

File: app/src/androidTest/java/ch/epfl/sdp/peakar/general/SettingsMapActivityTest.java [hunk 4]
```diff
@@ -19,21 +19,33 @@
...
 
 import ch.epfl.sdp.peakar.R;
+import ch.epfl.sdp.peakar.map.MapActivity;
```
Comment (ravifrancesco) [inline app/src/androidTest/java/ch/epfl/sdp/peakar/general/SettingsMapActivityTest.java:35]:
```suggestion
```

File: app/src/androidTest/java/ch/epfl/sdp/peakar/general/SettingsMapActivityTest.java [hunk 5]
```diff
@@ -19,21 +19,33 @@
...
+import static androidx.test.espresso.action.ViewActions.click;
 import static androidx.test.espresso.assertion.ViewAssertions.matches;
 import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
 import static androidx.test.espresso.matcher.ViewMatchers.withEffectiveVisibility;
 import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
```
Comment (ravifrancesco) [inline app/src/androidTest/java/ch/epfl/sdp/peakar/general/SettingsMapActivityTest.java:48]:
```suggestion
```

File: app/src/main/res/values-fr/strings.xml [hunk 6]
```diff
@@ -65,6 +65,8 @@
     <string name="position_display">"Position: (1%.2f, 2%.2f) "</string>
     <string name="date_display">Date: %s</string>
     <string name="setUsername">"Vous devez d'abord définir un nom d'utilisateur pour sauvegarder les pics découverts!"</string>
+    <string name="outsideOfDownloadedBox">"Vous etes en mode hors connexion et en-dehors de la région
+        télécharger. S'il vous plait éteignez le mode hors connexion dans les réglages pour utiliser l'application."</string>
```
Comment (bgabriel998) [inline app/src/main/res/values-fr/strings.xml:69]:
```suggestion
    <string name="outsideOfDownloadedBox">"Vous êtes en mode hors connexion et en-dehors de la région
        téléchargée. Veuillez quitter le mode hors connexion dans les réglages pour utiliser l'application."</string>
```

**Review by bgabriel998 — COMMENTED** [review]
Small changes requested. Otherwise PR is ready for merge

File: app/src/main/java/ch/epfl/sdp/peakar/general/SettingsMapActivity.java [hunk 7]
```diff
@@ -48,6 +50,7 @@
 
     private Button downloadButton;
     private View loadingView;
+    public static OSMMap osmMap;
```
Comment (ravifrancesco) [inline app/src/main/java/ch/epfl/sdp/peakar/general/SettingsMapActivity.java:53]: Why has this to be static?

**Review by ravifrancesco — CHANGES_REQUESTED** [review]
The code looks much cleaner now. Left a couple of suggestions and a question.

File: app/src/main/java/ch/epfl/sdp/peakar/general/SettingsMapActivity.java (hunk 7 above)
Comment (magangloff) [reply app/src/main/java/ch/epfl/sdp/peakar/general/SettingsMapActivity.java:53]: This was code that was copy-pasted. This is now resolved. It was made static to be used in the instrumented tests but I changed the public static to a private identifier.

**Review by ravifrancesco — APPROVED** [review]

**Review by bgabriel998 — APPROVED** [review]

**Pull request merged**